api/src/main/java/marquez/api/OpenLineageResource.java

```diff
@@ -61,20 +62,26 @@ public OpenLineageResource(
   @Consumes(APPLICATION_JSON)
   @Produces(APPLICATION_JSON)
   @Path("/lineage")
-  public void create(
-      @Valid @NotNull LineageEvent event, @Suspended final AsyncResponse asyncResponse)
+  public void create(@Valid @NotNull BaseEvent event, @Suspended final AsyncResponse asyncResponse)
       throws JsonProcessingException, SQLException {
-    openLineageService
-        .createAsync(event)
-        .whenComplete(
-            (result, err) -> {
-              if (err != null) {
-                log.error("Unexpected error while processing request", err);
-                asyncResponse.resume(Response.status(determineStatusCode(err)).build());
-              } else {
-                asyncResponse.resume(Response.status(201).build());
-              }
-            });
+    if (event instanceof LineageEvent) {
+      openLineageService
+          .createAsync((LineageEvent) event)
+          .whenComplete(
+              (result, err) -> {
+                if (err != null) {
+                  log.error("Unexpected error while processing request", err);
+                  asyncResponse.resume(Response.status(determineStatusCode(err)).build());
+                } else {
+                  asyncResponse.resume(Response.status(201).build());
+                }
+              });
+    } else {
+      log.warn("Unsupported event type {}. Skipping without error", event.getClass().getName());
+
+      // return serialized event
+      asyncResponse.resume(Response.status(200).entity(event).build());
```

Comment:
`200` status code is correct for the new OL events, and feel we should also return a `200` when accepting OL run events (as outlined by the OL [spec](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.yml#L23)). The semantics should be: "Return 200 to signify the OL event has been collected, and eventually will be processed." The `201` status code was never changed during the initial PoC phase of OL.

Reply:
The idea here was to distinguish between `RunEvent` that get saved into database (201 created) and other event types that do not affect application state. At the end, once Marquez will be capable of storing dataset and job events, it should return 201 for all the cases.

Reply:
@julienledem we'll want to write a proposal on how to handle `DatasetEvent`s and `JobEvent`s (see https://github.com/MarquezProject/marquez/issues/2544). For now, let's ensure the event can be accepted (but not stored).